### PR TITLE
Add constraint template and constraint YAML artifacts used by Microsoft.Kubernetes.Data mode policy definitions

### PR DESCRIPTION
**Pull request by RamyasreeChakka**
Add the OPA constraint framework constraint template and constraint YAML artifacts used by Gatekeeper V3, admission controller for AKS Engine. These artifacts are referred in built-in Azure policy definitions.

### CODE REVIEW

File: built-in-references/kubernetes/container-allowed-images/constraint.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: constraints.gatekeeper.sh/v1beta1
```
Comment (pilor) [inline built-in-references/kubernetes/container-allowed-images/constraint.yaml]: can the "kubernetes" folder be capitalized like "KubernetesService" for consistency?

File: built-in-references/kubernetes/container-allowed-images/constraint.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+kind: K8sAzureContainerAllowedImages
+metadata:
```
Comment (pilor) [inline built-in-references/kubernetes/container-allowed-images/constraint.yaml]: Does this content need a README describing how to enter the private preview like KubernetesService has?  If someone sees these files and wants to use them how do they get started?  The top level README just links to the language and a page about what AKS engine is

Comment (RamyasreeChakka) [reply built-in-references/kubernetes/container-allowed-images/constraint.yaml]: Good point. Thank you.
We are going public preview for AKS Engine, will point to the documentation in both cases. I will also mentioned AKS is in private preview and AKS Engine is in public preview.

File: built-in-references/kubernetes/container-allowed-images/constraint.yaml (hunk 1 above)
Comment (RamyasreeChakka) [reply built-in-references/kubernetes/container-allowed-images/constraint.yaml]: Sure

File: built-in-references/kubernetes/container-allowed-images/constraint.yaml (hunk 2 above)
Comment (RamyasreeChakka) [reply built-in-references/kubernetes/container-allowed-images/constraint.yaml]: Updated README.

**Pull request merged**